### PR TITLE
feat!: add options as parameter to constraints

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds options as part of the constraint callbacks so you have direct access to them when the models are being constrained, i.e. for conditionally changing the behavior.

This is only a breaking change if you use the current constraint types in TS integrations.

**Related issue(s)**
Part of solving https://github.com/asyncapi/modelina/issues/1475

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -55,7 +56,7 @@
     "no-empty-character-class": 2,
     "no-self-compare": 2,
     "valid-typeof": 2,
-    "no-unused-vars": 0,
+    "unused-imports/no-unused-imports": 2,
```
Comment (jonaslagoni) [inline .eslintrc:59]: Had to change lint rule because the old one did not understand `--fix` option

**Review by kennethaasan — APPROVED** [review]
LGTM

**Review by Samridhi-98 — APPROVED** [review]
LGTM 🎉

**Review by Ferror — APPROVED** [review]
PHP ok

**Comment by jonaslagoni** [comment]
Thanks for the review folks 🙇

**Pull request merged**